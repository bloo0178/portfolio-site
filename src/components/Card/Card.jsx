import React from "react";
import styles from "./Card.module.scss";

class Card extends React.Component {
  render() {
    return (
      <div className={styles.container}>
        <div>
          <h3>Title</h3>
        </div>
        <div>
          <p>
            Quo alii voluptua apeirian in. Ex modus noster nostrum sed. Qui no
            paulo platonem, has te ridens praesent.
          </p>
        </div>
        <div className={styles.links}>
          <a href="https://jeffbloom.io">Demo ></a>
          <a href="https://jeffbloom.io">Source ></a>
        </div>
      </div>
    );
  }
}

// validate proptypes here

export default Card;
