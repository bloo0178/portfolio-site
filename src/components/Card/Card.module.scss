@import "../../common/_variables";

.container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-content: center;
    background-color: $primary-background;
    color: $primary-text;
    width: 15rem;
    height: 15rem;
    padding: 1rem;
    margin: 1rem;
}

.links {
    a {
        margin-right: 1rem;
        margin-left: 1rem;
        color: $accent-text;
    }
}