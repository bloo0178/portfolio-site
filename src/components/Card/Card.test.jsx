import React from "react";
import { create } from "react-test-renderer";
import Card from "./Card";

const props = {
  demoURL: "demoURL.com",
  srcURL: "srcURL.com",
  title: "Test Title",
  description: `This is a project's description`, 
  tech: 'React, AWS, Serverless Framework'
};

describe("Card component", () => {
  const { demoURL, srcURL, title, description, tech } = props;
  test("it matches the snapshot", () => {
    const component = create(
      <Card
        demoURL={demoURL}
        srcURL={srcURL}
        title={title}
        description={description}
        tech={tech}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
