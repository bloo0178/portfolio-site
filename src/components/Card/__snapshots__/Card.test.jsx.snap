// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component it matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    <h3>
      Test Title
    </h3>
  </div>
  <div
    className="stack"
  >
    <p>
      Stack: 
      React, AWS, Serverless Framework
    </p>
  </div>
  <div
    className="links"
  >
    <a>
      Demo &gt;
    </a>
    <a>
      Source &gt;
    </a>
  </div>
</div>
`;
